core/types/block_ext_test.go

```diff
@@ -370,24 +373,19 @@ func TestNewBlockWithExtData(t *testing.T) {
 
 	tests := []struct {
 		name      string
-		header    func() *Header
+		header    *Header
 		txs       []*Transaction
-		uncles    func() []*Header
+		uncles    []*Header
 		receipts  []*Receipt
 		extdata   []byte
 		recalc    bool
 		wantBlock func() *Block
 	}{
 		{
-			name: "empty",
-			header: func() *Header {
-				header := &Header{}
-				SetHeaderExtra(header, &HeaderExtra{})
-				return header
-			},
+			name:   "empty",
```

Comment:
(optional) You could add another case where the header is a raw `&Header{}` to demonstrate that the getter will return a non-nil `&HeaderExtra{}`. Not sure of the utility other than someone reviewing the test learning that this is a valid use case (although there's definitely value in that).

Reply:
Done!